pkg/storage/node_liveness.go

```diff
@@ -229,7 +229,10 @@ func (nl *NodeLiveness) StartHeartbeat(
 	log.VEventf(ctx, 1, "starting liveness heartbeat")
 	retryOpts := base.DefaultRetryOptions()
 	retryOpts.Closer = stopper.ShouldQuiesce()
-	nl.heartbeatCallback = alive
+
+	nl.mu.Lock()
+	nl.mu.heartbeatCallback = alive
```

Comment:
Hrmm, my comment here disappeared. I wonder if `alive` should be passed to `NewNodeLiveness` instead of `StartHeartbeat`. That would avoid any locking issues.